Sources/SwiftExtensions/Cache.swift

```diff
@@ -40,7 +40,10 @@ package class Cache<Key: Sendable & Hashable, Result: Sendable> {
   ) async throws -> Result? {
     for (key, value) in storage {
       if keyPredicate(key) {
-        return try await transform(value.value)
+        let transformed = Task { try await transform(value.value) }
+        // Cache the transformed result.
+        storage[key] = transformed
+        return try await transformed.value
```

Comment:
Won't this re-apply the transform for every subsequent access?

Reply:
Indeed, this was wrong on so many different levels. Thanks for catching it in the review, this would have been a pain to debug.